apps/caracal.cpp

```diff
@@ -35,6 +35,9 @@ int main(int argc, char** argv) {
       ("L,log-level", "Minimum log level (trace, debug, info, warning, error, fatal)", cxxopts::value<string>()->default_value("info"))
       ("N,n-packets", "Number of packets to send per probe", cxxopts::value<int>()->default_value(std::to_string(config.n_packets)))
       ("P,max-probes", "Maximum number of probes to send (unlimited by default)", cxxopts::value<int>())
+      ("6,v6", "Specify that the IP version will be IPv6 (must use it when setting the source_ip)")
```

Comment:
Short 4/6 and long v4/v6 flags are redundant; just keep one. I'd vote for short 4/6 flags.

Reply:
The help message in unclear. Maybe something like "use the MAC address of the IPv6 gateway instead of the IPv4 gateway"? In which case we only need one flag, e.g. `--use-ipv6-gateway`.

Reply:
Ah or maybe I misunderstood and it's about the version of `--source-address`? In this case I would remove these flags and have `--source-address-v4` and `--source-address-v6` instead.